<?xml version="1.0"?>
<launch>
  <!-- The type of stop controller to use: 'proximity' or 'crash' -->
  <arg name="emergency_controller" default="none" />
 
  <include file="$(find husky_gazebo)/launch/husky_empty_world.launch">
    <arg name="world_name" value="$(find husky_highlevel_controller_v3)/worlds/singlePillar.world" />
    <arg name="laser_enabled" value="true" />
  </include>

  <!-- Main controller node -->
  <node name="husky_highlevel_controller_v3" pkg="husky_highlevel_controller_v3" type="husky_highlevel_controller_v3_node" output="screen">
    <rosparam command="load" file="$(find husky_highlevel_controller_v3)/config/default.yaml" />
  </node>

  <!-- Emergency stop controller nodes -->
  <group if="$(eval arg('emergency_controller') == 'proximity')">
    <node name="husky_proximity_stop_controller" pkg="husky_highlevel_controller_v3" type="husky_proximity_stop_controller_node" output="screen">
      <rosparam command="load" file="$(find husky_highlevel_controller_v3)/config/default.yaml" />
    </node>
  </group>
  
  <group if="$(eval arg('emergency_controller') == 'crash')">
    <node name="husky_crash_stop_controller" pkg="husky_highlevel_controller_v3" type="husky_crash_stop_controller_node" output="screen">
      <rosparam command="load" file="$(find husky_highlevel_controller_v3)/config/default.yaml" />
    </node>
  </group>

  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find husky_highlevel_controller_v3)/rviz/husky_highlevel_controller.rviz" />
</launch>
